src/s3/util/fileViewerManager.ts

```diff
@@ -21,25 +21,76 @@ import { localize } from '../../shared/utilities/vsCodeUtils'
 const SIZE_LIMIT = 4 * Math.pow(10, 6)
 
 export class S3FileViewerManager {
-    private cacheArns: Set<string>
-    private activeTabs: Map<string, S3Tab>
-    private window: typeof vscode.window
     private outputChannel: OutputChannel
-    private commands: Commands
-    private tempLocation: string | undefined
+    private promptOnEdit = true
+    //onDidChange to trigger refresh of contents on the document provider
+    private _onDidChange = new vscode.EventEmitter<vscode.Uri>()
+    public get onDidChange(): vscode.Event<vscode.Uri> {
+        return this._onDidChange.event
+    }
 
     public constructor(
-        cacheArn: Set<string> = new Set<string>(),
-        window: typeof vscode.window = vscode.window,
-        commands = Commands.vscode(),
-        tempLocation?: string
+        private cacheArns: Set<string> = new Set<string>(),
+        private window: typeof vscode.window = vscode.window,
+        private commands = Commands.vscode(),
+        private _tempLocation?: string,
+        private activeTabs: Map<string, S3Tab> = new Map<string, S3Tab>()
     ) {
-        this.cacheArns = cacheArn
-        this.activeTabs = new Map<string, S3Tab>()
-        this.window = window
-        this.commands = commands
-        this.tempLocation = tempLocation
         this.outputChannel = ext.outputChannel
+
+        vscode.workspace.onDidSaveTextDocument(async savedTextDoc => {
+            if (this.activeTabs.has(savedTextDoc.uri.fsPath)) {
```

Comment:
Return early?

Reply:
It was just added to avoid using `activeTab!` since the compiler would say it may be undefined, but because of the conditional in line 42 it can be assumed that is not the case

Reply:
Fair enough. TypeScript doesn't work with `has` on sets/maps for some reason. Guess they never added the type predicate.